File: domain/src/main/java/com/arturogutierrez/openticator/domain/category/repository/CategoryDataStore.java
package com.arturogutierrez.openticator.domain.category.repository;

import com.arturogutierrez.openticator.domain.account.model.Account;
import com.arturogutierrez.openticator.domain.category.model.Category;
import java.util.List;
import rx.Observable;

public interface CategoryDataStore {

  Observable<Category> add(Category category);

  Observable<Category> addAccount(Category category, Account account);

  Observable<List<Category>> getCategories();
}

File: domain/src/main/kotlin/com/arturogutierrez/openticator/domain/category/repository/CategoryDataStore.kt
package com.arturogutierrez.openticator.domain.category.repository

import com.arturogutierrez.openticator.domain.account.model.Account
import com.arturogutierrez.openticator.domain.category.model.Category
import io.reactivex.Flowable
import io.reactivex.Single

interface CategoryDataStore {

  fun add(category: Category): Single<Category>

  fun addAccount(category: Category, account: Account): Single<Category>

  val categories: Flowable<List<Category>>
}
